doc/tutorial.rst

```diff
@@ -89,117 +97,140 @@ We will use the following very simple Dockerfile:
-the root of the Docker context directory. (We will address the context
-directory in more detail later.)
+.. note::
+   :code:`ch-image` prints information about the build process while in
+   progress. While not demonstrated above, yellow is used for :code:`ch-image`
+   internal chatter.
 
-The two arguments here are a tag for the Docker image and the context
-directory, which in this case is the current directory.
+This image and the :code:`centos:8` base image used to build it are now
+visible in Charliecloud's builder storage:
 
 ::
 
-   $ ch-build -t hello .
-   Sending build context to Docker daemon  5.632kB
-   Step 1/4 : FROM centos:8
-    ---> 0d120b6ccaa8
-   [...]
-   Step 4/4 : RUN touch /usr/bin/ch-ssh
-    ---> eb5f42d5bb54
-   Successfully built eb5f42d5bb54
-   Successfully tagged hello:latest
-
-Note that Docker prints each step of the Dockerfile as it's executed.
-
-:code:`ch-build` and many other Charliecloud commands wrap various privileged
-:code:`docker` commands. Thus, you will be prompted for a password to escalate
-as needed. Note however that most configurations of :code:`sudo` don't require
-a password on every invocation, so privileged commands may be running even if
-you're not prompted for a password.
-
-Share image and other standard Docker stuff
+  $ ch-image list
+  centos:8
+  hello
+
+.. note::
+   Charliecloud uses image :code:`name:tag` to refer to images instead
+   of a computed hash like some other image builders.
+
+Sharing images
 -------------------------------------------
 
-If needed, the Docker image can be manipulated with standard Docker commands.
-In particular, image sharing using a public or private Docker Hub repository
-can be very useful.
+Charliecloud images in builder storage are just directories and can be
+exported as SquashFS filesystems or tar archives via :code:`ch-convert`.
+
+SquashFS:
+::
 
+  $ ch-convert hello /var/tmp/hello.sqfs
+  input:   ch-image  hello
+  output:  squash    /var/tmp/hello.sqfs
+  packing ...
+  [...]
+  done
+  $ ls -ld /var/tmp/hello.sqfs 
+  -rw-r--r-- 1 root root 83288064 Nov 15 12:07 /var/tmp/hello.sqfs
+    
+tarball:
 ::
 
-  $ sudo docker images
-  REPOSITORY  TAG      IMAGE ID      CREATED        SIZE
-  centos      8        0d120b6ccaa8  2 months ago   215MB MB
-  hello       latest   eb5f42d5bb54  5 minutes ago  235MB MB
-  $ sudo docker push  # FIXME
+  $ ch-convert hello /var/tmp/hello.tgz
+  input:   ch-image  hello
+  output:  tar       /var/tmp/hello.tgz
+  exporting ...
+  done
+  $ ls -ld /var/tmp/hello.tgz
+  -rw-rw-r-- 1 heasterday heasterday 86122450 Nov 15 15:23 /var/tmp/hello.tgz
 
-Running the image with Docker is not generally useful, because Docker's
-run-time environment is significantly different than Charliecloud's, but it
-can have value when debugging Charliecloud.
+Charliecloud can also convert images between the two formats.
 
+SquashFS to tarball:
```

Comment:
I wonder if converting images between formats is too much detail for a tutorial? I do like the form we settled on for the workshop with some of the things relegated to "advanced topics" at the end.

Reply:
I think it's good to be aware that the functionality exists and don't feel that it adds too much complexity. I defer to you though.